abdm_python_integrator/utils.py

```diff
@@ -0,0 +1,78 @@
+import functools
```

Comment:
Could consider moving this to existing utils directory.

Reply:
This is a common util and is getting used for M2 and M3 as well.

Reply:
Sure, what's that to do with the directory structure?

Reply:
The exisiting `utils` directory is part of `abha` . This `utils.py` is planned to be at root `abdm_python_integrator` level and is used by M2 (hiu) and M3(hip) as well.
We will have `hiu` and `hip` folders (as django apps) at the same level as `abha`. The idea here is to keep these three isolated from each other to make it convenient for customer to use/drop as per their requirement.

Reply:
I don't get it fully, but I think this is a safe one to revisit latter when we do add other directories.